kernel/src/process_loading.rs

```diff
@@ -496,6 +535,10 @@ pub struct SequentialProcessLoaderMachine<'a, C: Chip + 'static, D: ProcessStand
     storage_policy: &'static dyn ProcessStandardStoragePermissionsPolicy<C, D>,
     /// Current mode of the loading machine.
     state: OptionalCell<SequentialProcessLoaderMachineState>,
+    /// Current client of the loading machine.
+    active_client: OptionalCell<SequentialProcessLoaderMachineActiveClient>,
+    /// Lookup tables for process binaries addresses and sizes
+    process_binaries_metadata: OptionalCell<ProcessBinaryMetadata>,
```

Comment:
It seems like `process_binaries_metadata` can just be allocated on the stack and returned from `scan_flash_for_app_binaries()`.